engines/scumm/he/sound_he.cpp

```diff
@@ -649,12 +650,16 @@ void SoundHE::playHESound(int soundID, int heOffset, int heChannel, int heFlags,
 
 		_mixer->stopHandle(_heSoundChannels[heChannel]);
 		if (compType == 17) {
-			Audio::AudioStream *voxStream = Audio::makeADPCMStream(&memStream, DisposeAfterUse::NO, size, Audio::kADPCMMSIma, rate, (flags & Audio::FLAG_STEREO) ? 2 : 1, blockAlign);
+			int nChan = (flags & Audio::FLAG_STEREO) ? 2 : 1;
+			Audio::AudioStream *voxStream = Audio::makeADPCMStream(&memStream, DisposeAfterUse::NO, size, Audio::kADPCMMSIma, rate, nChan, blockAlign);
 
 			// FIXME: Get rid of this crude hack to turn a ADPCM stream into a raw stream.
 			// It seems it is only there to allow looping -- if that is true, we certainly
 			// can do without it, using a LoopingAudioStream.
 
+			if (_heChannel[heChannel].timer)
+				_heChannel[heChannel].timer = size / nChan * samplesPerBlock / blockAlign * 1000 / rate;
```

Comment:
```suggestion
				_heChannel[heChannel].timer = (size / nChan) * (samplesPerBlock / blockAlign) * (1000 / rate);
```
This helps make the calculation a bit more readable

We could also write it like this:

    _heChannel[heChannel].timer = (size * samplesPerBlock * 1000) / (rate * blockAlign * nChan);

Reply:
Unfortunately, because this is integer arithmetic, adding those parentheses in your suggested change would cause the calculation to come out to 0 every time. For illustration, here are some example values I recorded:

```
size=28448
nChan=2
samplesPerBlock=1017
blockAlign=512
rate=22050
```

The second suggestion was how I had initially thought to do it too, but the numerator will overflow 32-bit integers. It should work though if we did something like this:

```
(int)(((int64)size * samplesPerBlock * 1000) / (rate * blockAlign * nChan));
```

I went with the order/formatting I did because it preserved precision without requiring 64-bit or floating-point arithmetic. But I can change it to the 64-bit version. (It would be guaranteed the most precise version too).

Reply:
What about this then? This should work with 32-bit math:
     _heChannel[heChannel].timer = 1000 * ((size * samplesPerBlock) / (rate * blockAlign * nChan));

Reply:
That would unfortunately cause all numbers to be rounded down to the nearest 1000. It causes some dialog to get cut short.

Reply:
OK. It would be preferable to use the 64-bit integer calculation that you posted above, then. It feels safer this way, IMHO.